types/audit-details.d.ts

```diff
@@ -5,63 +5,166 @@
-        /** The raw timestamp of this frame, in microseconds. */
+  module LH.Audit {
+    export type Details =
+      Details.CriticalRequestChain |
+      Details.Diagnostic |
+      Details.Filmstrip |
+      Details.Opportunity |
+      Details.Screenshot |
+      Details.Table;
+
+    // Details namespace.
+    export module Details {
+      export interface CriticalRequestChain {
+        type: 'criticalrequestchain';
+        longestChain: {
+          duration: number;
+          length: number;
+          transferSize: number;
+        };
+        chains: Audit.SimpleCriticalRequestNode;
+      }
+
+      export interface Filmstrip {
+        type: 'filmstrip';
+        scale: number;
+        items: {
+          /** The relative time from navigationStart to this frame, in milliseconds. */
+          timing: number;
+          /** The raw timestamp of this frame, in microseconds. */
+          timestamp: number;
+          /** The data URL encoding of this frame. */
+          data: string;
+        }[];
+      }
+
+      export interface Opportunity {
+        type: 'opportunity';
+        overallSavingsMs: number;
+        overallSavingsBytes?: number;
+        headings: OpportunityColumnHeading[];
+        items: OpportunityItem[];
+        diagnostic?: Diagnostic;
+      }
+
+      export interface Screenshot {
+        type: 'screenshot';
         timestamp: number;
-        /** The data URL encoding of this frame. */
         data: string;
-      }[];
-    }
+      }
 
-    export interface Screenshot {
-      type: 'screenshot';
-      timestamp: number;
-      data: string;
-    }
+      export interface Table {
+        type: 'table';
+        headings: TableColumnHeading[];
+        items: TableItem[];
+        summary?: {
+          wastedMs?: number;
+          wastedBytes?: number;
+        };
+        diagnostic?: Diagnostic;
+      }
 
-    export interface Opportunity {
-      type: 'opportunity';
-      overallSavingsMs: number;
-      overallSavingsBytes?: number;
-      headings: OpportunityColumnHeading[];
-      items: OpportunityItem[];
-    }
+      /**
+       * A details type that is not rendered in the final report; usually used
+       * for including diagnostic information in the LHR. Can contain anything.
+       */
+      export interface Diagnostic {
+        type: 'diagnostic';
+        [p: string]: any;
+      }
 
-    export interface CriticalRequestChain {
-      type: 'criticalrequestchain';
-      longestChain: {
-        duration: number;
-        length: number;
-        transferSize: number;
-      };
-      chains: Audit.SimpleCriticalRequestNode;
-    }
+      /** Possible types of values found within table items. */
+      type ItemValueTypes = 'bytes' | 'code' | 'link' | 'ms' | 'node' | 'numeric' | 'text' | 'thumbnail' | 'timespanMs' | 'url';
 
-    // Contents of details below here
+      // TODO(bckenny): unify Table/Opportunity headings and items on next breaking change.
```

Comment:
`TableColumnHeading` and `OpportunityColumnHeading` are very close but not close enough. I think we just didn't get around to making them the same :S

Something for v5 :)